semtype.lisp

```diff
@@ -0,0 +1,83 @@
+;; Data structure for representing ULF semantic types
+
+(in-package :ulf-lib)
+
+;; Class representing a ULF semantic type
+(defclass semtype ()
+  ((domain
+     :initarg :domain
+     :accessor domain)
+   (range
+     :initarg :range
+     :initform nil
+     :accessor range)
+   (exponent
+     :initarg :ex
+     :initform 1
+     :accessor ex)
+   (subscript
+     :initarg :subscript
+     :initform nil
+     :accessor subscript)
+   (tense?
```

Comment:
I usually reserve symbols ending in a question mark for functions that return `t` or `nil` for checking a property. e.g. `noun?` for checking if something is a noun. I think it would be better to call this just `tense` or perhap `tensed`.